app/models/task.py

```diff
@@ -1,5 +1,63 @@
 from app import db
+from sqlalchemy import sql
+from datetime import datetime
+import requests
+import os
 
 
 class Task(db.Model):
     task_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String, nullable=False)
+    description = db.Column(db.String, nullable=False)
+    completed_at = db.Column(db.DateTime, default=None)
+    goal_id = db.Column(db.Integer, db.ForeignKey("goal.goal_id"))
+    goal = db.relationship("Goal", back_populates="tasks")
+
+    def to_dict(self):
+        task = {
+            "id": self.task_id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.completed_at is not None,
+        }
+        if self.goal_id:
+            task["goal_id"] = self.goal_id
+        return task
+
+    def update(self, **kwargs):
+        for key in kwargs:
+            if key in ("title", "description"):
+                setattr(self, key, kwargs[key])
+            if key == "is_complete":
+                self.mark_complete(when=kwargs[key] or False)
+            if key == "completed_at":
+                self.mark_complete(when=kwargs[key])
+            if key == "goal_id":
+                self.add_goal(self)
```

Comment:
The conciseness of this pattern really shines here. I like it! Your code has honestly made me appreciate Python a lot more. :3

Reply:
i actually felt like this was one of the worse designs in my api. i want to revisit and find another solution